### DISCUSSION TITLE
How to add proxy for target url

### DISCUSSION THREAD

**Discussion by bboylilshaw** [chimurai/http-proxy-middleware#816] [Q&A] [answered]
Hi 
I am currently using **http-proxy-middleware** in my express app to route all the frontend ajax requests to my backend endpoint, but my backend endpoint is behind a gateway, I need to use a proxy to call my backend.
I looked through the document, and didn't find anything about setting proxy (needs authentication) for my target url.
can anyone help?

**Comment by chimurai** [comment, marked as answer]
Think this is what you are looking for:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/corporate-proxy.md#readme
